### PR TITLE
Banking System

### PR DESCRIPTION
**Pull request by sristtii**
simple banking system overview

### CODE REVIEW

File: sristii/Main.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+//oops overview
+//abstraction(hiding the actual implementation details and exposing onlu functionality)
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:7]: Please add java doc if needed

File: sristii/Main.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:5]: Please split the classes into several classes (from the file perspective)

File: sristii/Main.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        System.out.println("Welcome to banking System");
+    }
+}
+
+//idea of interface
+//interface used to achieve abstraction,multiple inheritance
+//multiple inheritance not supported in java so
+//can be acheived from interface
+interface AccountManagement {
+    void change_id(int newId);
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:22]: Please use camel case(changeId)

File: sristii/Main.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        this.name = name;
+        this.balance = balance;
+        this.age = age;
+    }
+
+    // Implementing the abstract method from AccountOperations
+    @Override
+    //hiding the implementation here
+    public void accountSummary() {
+        System.out.println("id: " + accId);
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:51]: Messageformat - https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/MessageFormat.html

Comment (vivek-ravichandran-trustt) [reply sristii/Main.java:51]: Messageformat.format("{} printed  - {} success",a,b)

File: sristii/Main.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        List<Integer> accountIds = accounts.stream()
+                .map(BankAccount::getAccId)
+                .sorted()//takes out accountids
+                .collect(Collectors.toList());
+        //using streams to store names
+        List<String> accountOwners = accounts.stream()
+                .map(BankAccount::getName)
+                .collect(Collectors.toUnmodifiableList());
+        //lets try to sort out ages
+        List<Integer> agesofall = accounts.stream()
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:185]: Please write this function using for loop

File: sristii/Main.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        System.out.println("ages of people: " + agesofall);
+        System.out.println("names of ppl: " + accountOwners);
+        System.out.println("account id numbers: "+accountIds);
+        //going through polymorphism
+        List<BankAccount> savings = new ArrayList<>();
+        savings.add(new Savings(123, "sristtii", 2345.23, 20,2.3));
+        savings.add(new Savings(234, "harini", 3427.78, 21,3.5));
+        savings.add(new Savings(345, "tas", 2425.32, 23,5.7));
+        //calling accountsummary for savings
+        for(BankAccount i:savings){
```
Comment (vivek-ravichandran-trustt) [inline sristii/Main.java:205]: try this with stream

File: sristii/Main.java (hunk 2 above)
Comment (vivek-ravichandran-trustt) [reply sristii/Main.java:5]: Changes handled

File: sristii/Main.java (hunk 3 above)
Comment (sristtii) [reply sristii/Main.java:22]: Changed Method Names into CamelCase

**Pull request merged**